01-dsa/01-dsa.mk

```diff
@@ -0,0 +1,6 @@
+SRCDIR=$(BASEDIR)/01-dsa
+
+mod1:
+	$(CC) $(SRCDIR)/$@.c -o $(BINDIR)/$@.elf
+
+all: mod1
```

Comment:
target `all` should be defined in top-level makefile and just reference targets in these distributed makefiles